src/Tickets/Commerce/Gateways/Stripe/Payment_Intent_Handler.php

```diff
@@ -252,4 +255,47 @@ static function ( $item ) {
 		 */
 		return apply_filters( 'tec_tickets_commerce_stripe_update_payment_intent_metadata', $metadata, $order, $payment_intent );
 	}
+
+	/**
+	 * Get the description for the payment intent.
+	 *
+	 * @since TBD
+	 *
+	 * @param \WP_Post $order The Order data.
+	 * @param array    $data The purchase data received from the front-end.
+	 * @param array    $body The body used to update the payment intent.
+	 * @param array    $payment_intent The Payment intent.
+	 *
+	 * @return array
+	 */
+	protected function get_payment_intent_description( \WP_Post $order, $data, $body, array $payment_intent ) {
+		$purchaser_name = $order->purchaser_name;
+
+		if ( is_user_logged_in() ) {
+			$user           = wp_get_current_user();
+			$purchaser_name = $user->get( 'first_name' ) ? $user->get( 'first_name' ) . ' ' . $user->get( 'last_name' ) : $user->get( 'display_name' );
+		}
+
+		$tickets_in_order           = implode( ', ', array_unique( array_filter( wp_list_pluck( $order->items, 'ticket_id' ) ) ) );
+		$events_in_order            = array_unique( array_filter( wp_list_pluck( $order->items, 'event_id' ) ) );
+		$post_id                    = current( $events_in_order );
+		$post                       = get_post( $post_id );
+		$post_labels                = get_post_type_labels( get_post_type_object( $post->post_type ) );
+		$payment_intent_description = sprintf( '[%1$s: %2$s] [Ticket: %3$s] [Order: %4$s] %5$s - %6$s - %7$s', $post_labels->singular_name, $post_id, $tickets_in_order, $order->ID, $body['metadata']['event_name'], $body['metadata']['ticket_names'], $purchaser_name );
```

Comment:
Let's lay this out a little nicer, pull the correct terminology for "Ticket" and make sure "Order" is translatable:
```suggestion
		$payment_intent_description = sprintf(
			'[%1$s: %2$s] [%3$s: %4$s] [%5$s: %6$s] %7$s - %8$s - %9$s',
			$post_labels->singular_name,
			$post_id,
			tribe_get_ticket_label_singular( 'stripe_payment_intent_description' ),
			$tickets_in_order,
			__( 'Order', 'event-tickets' ),
			$order->ID,
			$body['metadata']['event_name'],
			$body['metadata']['ticket_names'],
			$purchaser_name
		);
```

Reply:
Thanks @codingmusician ! I have applied your suggestion.